oak_runtime/src/node/http/mod.rs

```diff
@@ -465,56 +528,146 @@ fn get_oak_label(req: &HttpRequest) -> Result<Label, OakStatus> {
             err
-        );
-        OakStatus::ErrInvalidArgs
+        ))
     })?;
-    serde_json::from_str(&label_str).map_err(|err| {
-        warn!("Could not parse HTTP label: {}", err);
-        OakStatus::ErrInvalidArgs
-    })
+    serde_json::from_str(&label_str)
+        .map_err(|err| HttpError::RequestLabel(format!("Could not parse HTTP label: {}", err)))
 }
 
-fn parse_protobuf_label(base64_protobuf_label: &[u8]) -> Result<Label, OakStatus> {
+fn parse_protobuf_label(base64_protobuf_label: &[u8]) -> Result<Label, HttpError> {
     let protobuf_label = base64::decode(base64_protobuf_label).map_err(|err| {
-        warn!("Could not decode Base64 HTTP label: {}", err);
-        OakStatus::ErrInvalidArgs
+        HttpError::RequestLabel(format!("Could not decode Base64 HTTP label: {}", err))
+    })?;
+    Label::decode(&protobuf_label[..])
+        .map_err(|err| HttpError::RequestLabel(format!("Could not parse HTTP label: {}", err)))
+}
+
+/// Checks that the input signature (containing the signed challenge and the corresponding public
+/// key) is valid. If the signature is valid, this function returns the public key, otherwise
+/// returns an [`HttpError`].
+fn verify_json_challenge(signature: &[u8]) -> Result<Vec<u8>, HttpError> {
+    let signature = parse_json_signed_challenge(signature.to_vec()).map_err(|err| {
+        HttpError::IdentityVerification(format!(
+            "Could not parse json formatted signed challenge: {:?}",
+            err
+        ))
+    })?;
+    verify_signed_challenge(signature)
+}
+
+/// Tries to parse the signed challenge retrieved from the HTTP request into an instance of
+/// [`SignedChallenge`]. If not successful, returns an [`HttpError`].
+fn parse_json_signed_challenge(bytes: Vec<u8>) -> Result<SignedChallenge, HttpError> {
+    let signature_str = String::from_utf8(bytes).map_err(|err| {
+        HttpError::IdentityVerification(format!("Could not parse signed challenge: {:?}", err))
     })?;
-    Label::decode(&protobuf_label[..]).map_err(|err| {
-        warn!("Could not parse HTTP label: {}", err);
-        OakStatus::ErrInvalidArgs
+    serde_json::from_str(&signature_str).map_err(|err| {
+        HttpError::IdentityVerification(format!("Malformed signed challenge: {:?}", err))
     })
 }
 
-struct HttpResponseIterator {
+/// Checks that the input signature (containing the signed challenge and the corresponding public
+/// key) is valid. If the signature is valid, this function returns the public key, otherwise it
+/// returns an [`HttpError`].
+fn verify_protobuf_challenge(base64_signature: &[u8]) -> Result<Vec<u8>, HttpError> {
+    let signature = base64::decode(base64_signature).map_err(|err| {
+        HttpError::IdentityVerification(format!(
+            "Could not decode Base64 signed challenge: {}",
+            err
+        ))
+    })?;
+    let signature = SignedChallenge::decode(&signature[..]).map_err(|err| {
```

Comment:
Why is `let signature` defined 2 times? Could you please rename the first one into `signature_bytes` or similar?